pkg/sriov/sriov.go

```diff
@@ -245,8 +259,14 @@ func (s *sriovManager) ReleaseVF(conf *sriovtypes.NetConf, podifName string, cid
 				return fmt.Errorf("failed to parse original effective MAC address %s: %v", conf.EffectiveMAC, err)
 			}
 
-			if err = s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
-				return fmt.Errorf("failed to restore original effective netlink MAC address %s: %v", hwaddr, err)
+			if originalErr := s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); originalErr != nil {
```

Comment:
Just a thing that pops to mind,
if `conf.EffectiveMAC` is all zeroes then set  a random MAC, else restore to the saved value.
then we would not trigger netlink commands twice.

if  LinkSetHardwareAddr  fails  from other reasons, there may be an issue with the system,  so the second call will also fail.

i dont think we need to worry about collisions